prime-router/src/main/kotlin/cli/LookupTableCommands.kt

```diff
@@ -612,30 +674,287 @@ class LookupTableCompareMappingCommand : GenericLookupTableCommand(
+                                valueSet.expansion.contains.map { test -> // add every test in the oid valueset
+                                    mapOf( // assemble map of test data
+                                        ObservationMappingConstants.TEST_CODE_KEY to test.code,
+                                        ObservationMappingConstants.TEST_CODESYSTEM_KEY to // coerce to our values
+                                            ObservationMappingConstants.TEST_CODESYSTEM_MAP.getOrDefault(
+                                                test.system,
+                                                test.system
+                                            ),
+                                        ObservationMappingConstants.TEST_OID_KEY to valueSet.idElement.idPart,
+                                        ObservationMappingConstants.TEST_NAME_KEY to valueSet.name,
+                                        ObservationMappingConstants.TEST_DESCRIPTOR_KEY to test.display,
+                                        ObservationMappingConstants.TEST_STATUS_KEY to valueSet.status.toString()
+                                            .lowercase().replaceFirstChar(Char::titlecase),
+                                        ObservationMappingConstants.TEST_VERSION_KEY to test.version.split('/').last()
+                                    )
+                                }
+                            }
+                        } catch (e: DataFormatException) {
+                            throw PrintMessage("Error encountered parsing update for oid '$oid':\n{$e.message}", true)
+                        }
+                    }
+                }.mapValues { it.value.await() } // un-defer all values after starting coroutines
+            }
+        }
+
+        /**
+         * Generate a new observation mapping table with updated test data [updateOIDMap] using condition data from an
+         * existing observation mappings [tableOIDMap]; both maps grouping mappings by member oid
+         * @return updated observation mapping table as a list of maps
+         */
+        fun syncMappings(
+            tableOIDMap: Map<String, List<Map<String, String>>>,
+            updateOIDMap: Map<String, List<Map<String, String>>>,
+        ): List<Map<String, String>> {
+            return updateOIDMap.map { update -> // process every oid test group update
+                val conditionData = tableOIDMap[update.key]!![0].filterKeys {
+                    it in ObservationMappingConstants.CONDITION_KEYS // fetch existing condition data for this oid
+                }
+                update.value.map { it + conditionData } // add condition data to each test
+            }.flatten() + tableOIDMap.filterKeys { it !in updateOIDMap.keys }.values.flatten() // flatten + add carryover
+        }
+    }
+
+    override fun run() {
+        val tableData = if (inputFile != null) { // Start with data from a file
```

Comment:
What do you think about breaking some of this code up into more discrete pieces? Feel like there are a few sections in this that could be split out (tableData, httpclient). What you have works but I think it might be a little hard to read/follow.

Reply:
Great suggestion, I moved the code for building the api client and the table oid map into discrete functions.

I also added test coverage for the latter.